opennms-services/src/main/java/org/opennms/protocols/jmx/connectors/Jsr160ConnectionFactory.java

```diff
@@ -222,11 +223,23 @@ private static JMXServiceURL getUrl(InetAddress address, int port, String protoc
             // Create an JMXMP connector client and
             // connect it to the JMXMP connector server
             //
-            url = new JMXServiceURL(protocol, InetAddressUtils.str(address), port, urlPath);
+            url = new JMXServiceURL(protocol, toUrlIpAddress(address), port, urlPath);
         } else {
             // Fallback, building a URL for RMI
-            url = new JMXServiceURL("service:jmx:" + protocol + ":///jndi/" + protocol + "://" + InetAddressUtils.str(address) + ":" + port + urlPath);
+            url = new JMXServiceURL("service:jmx:" + protocol + ":///jndi/" + protocol + "://" + toUrlIpAddress(address) + ":" + port + urlPath);
         }
         return url;
-    }    
+    }
+
+    /**
+     * Method that wraps IPv6 addresses in square brackets so that they are parsed
+     * correctly by the {@link JMXServiceURL} class.
+     */
+    private static String toUrlIpAddress(InetAddress addr) {
```

Comment:
I had asked about that when he talked about this in scrum, but the problem is that the square brackets are only for URLs, not general display of IP addresses.  He could add a specific "url-friendly" method to `InetAddressUtils`, I suppose, but it's not appropriate for `str`.